source/server/listener_impl.cc

```diff
@@ -467,6 +467,18 @@ void ListenerImpl::buildAccessLog() {
 void ListenerImpl::buildInternalListener() {
   if (config_.address().has_envoy_internal_address()) {
     internal_listener_config_ = std::make_unique<Network::InternalListenerConfig>();
+    if (config_.has_connection_balance_config() || config_.enable_mptcp() ||
+        config_.has_enable_reuse_port() || config_.has_freebind() ||
+        config_.has_tcp_backlog_size() || config_.has_tcp_fast_open_queue_length() ||
+        config_.has_transparent()) {
```

Comment:
`|| config_.enable_mptcp()`. And maybe `has_api_listener()`? 

For all of the BoolValue's, I think it's still valid if it is present, but set to false. 

This validation is going to be a big mess. Maybe it's not worth doing. Any guidance from @envoyproxy/api-shepherds  on what validation to do for mutually-incompatible options?

Reply:
re: it's still valid if it is present. Right, Let me fix immediately.

Regarding the high level validation, I remember we opened an issue to move the tcp only options to a dedicated tcp listener config, though there was no solid move. That would be the long term fix. 
Another fact is that Udp listener doesn't validate tcp options, just ignore.

I will add the quick fix on the validation here and leave api shephereds to the final call.